### PR TITLE
new flags: --client-only & --client-language

### PR DESCRIPTION
**Pull request by aastein**
This PR adds two command line flags to Wag, `--client-only` and `--client-language`

**`--client-only`** When used wag does not generate server code

**`--client-language`** Can be `go` or `js`. When used wag will generate only the client of the specified language. 
 
**Both flags can be used together or separately.**  When using both flags like so `--client-only --client-language js`, no golang models are generated.

- [x] Run `make build`
- [x] Run `make generate`
- [x] Update the current version in the `/VERSION` file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,49 +23,49 @@ import (
...
+	// flag values
+	swaggerFile    *string
+	goPackageName  *string
+	outputPath     *string
+	jsModulePath   *string
+	versionFlag    *bool
+	clientOnly     *bool
+	clientLanguage *string
+
+	// dervied values
```
Comment (ghirsch1) [inline main.go:37]:
```suggestion
	// derived values
```

File: main.go [hunk 2]
```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
 	moduleName = strings.TrimSpace(moduleName)
 	return fmt.Sprintf("%v/%v", moduleName, outputPath)
 }
+
+// validate validates the configuration and determines the goPackagePath for the generated code
+func (c *config) validate() error {
```
Comment (ghirsch1) [inline main.go:222]: nit, but could rename to `validateAndSetDerivedFields` to convey that it's not just validating

File: main.go [hunk 3]
```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
...
+		c.goPackagePath = getModulePackagePath(goPath, *c.outputPath)
+		*c.goPackageName = getModulePackageName(modFile, *c.outputPath)
+	}
+
+	// parse flags determing if client/server code is generated
+	c.generateServer = true
+	c.generateGoClient = true
+	c.generateJSClient = true
+	// only generate client
+	if c.clientOnly != nil && *c.clientOnly == true {
```
Comment (ghirsch1) [inline main.go:253]: nit, we often use this style
```suggestion
	if swag.BoolValue(c.clientOnly) {
```

File: main.go [hunk 4]
```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
...
+		c.generateServer = false
+	}
+
+	// only generate client of specific language
+	if c.clientLanguage != nil && len(*c.clientLanguage) > 0 {
```
Comment (ghirsch1) [inline main.go:258]: same nit
```suggestion
	if swag.StringValue(c.clientLanguage) != "" {
```

File: main.go [hunk 5]
```diff
@@ -176,3 +217,72 @@ func getModulePackageName(modFile *os.File, outputPath string) string {
...
+		default:
+			return fmt.Errorf("client-language must be one of \"go\" or \"js\"")
+		}
+	} else {
+		c.generateGoClient = true
+		c.generateJSClient = true
+	}
+
+	c.generateGoModels = c.generateServer || c.generateGoClient
+	if c.generateJSClient && (c.jsModulePath == nil || *c.jsModulePath == "") {
```
Comment (ghirsch1) [inline main.go:278]:
```suggestion
	if c.generateJSClient && swag.StringValue(c.jsModulePath) == "" {
```

**Review by ghirsch1 — APPROVED** [review]
Looks awesome!

File: main.go (hunk 2 above)
Comment (aastein) [reply main.go:222]: just gonna call it `setDerivedFields` because it returns an error so validation is somewhat implied.

**Pull request merged**